extensions/strategies/trendline/strategy.js

```diff
@@ -1,113 +1,66 @@
-let math = require('mathjs')
-  , trend = require('trend')
-  , z = require('zero-fill')
-  , n = require('numbro')
+var z = require('zero-fill')
   , stats = require('stats-lite')
+  , math = require('mathjs')
   , ema = require('../../../lib/ema')
   , Phenotypes = require('../../../lib/phenotype')
-var oldgrowth = 1
+  , regression = require('regression')
+  , trend = require('trend')
+  , n = require('numbro')
 
 module.exports = {
   name: 'trendline',
-  description: 'Calculate a trendline and trade when trend is positive vs negative.',
-
+  description: 'Buy on positive trendline above 1.1 Make sure to set --days when backtesting period.',
   getOptions: function () {
-    this.option('period', 'period length', String, '30s')
-    this.option('periodLength', 'period length', String, '30s')
-    this.option('lastpoints', 'Number of trades for short trend average', Number, 100)
-    this.option('avgpoints', 'Number of trades for long trend average', Number, 1000)
-    this.option('lastpoints2', 'Number of trades for short trend average', Number, 10)
-    this.option('avgpoints2', 'Number of trades for long trend average', Number, 100)
-    this.option('min_periods', 'Basically avgpoints + a BUNCH of more preroll periods for anything less than 5s period', Number, 15000)
-    this.option('markup_sell_pct', 'test', Number, 0)
-    this.option('markdown_buy_pct', 'test', Number, 0)
+    this.option('period', 'period length, set poll trades to 100ms, poll order 1000ms. Same as --period_length', String, '1h')
+    this.option('period_length', 'period length, set poll trades to 100ms, poll order 1000ms. Same as --period', String, '1h')
+    this.option('avgpoints', 'Trades for array 2 to be calculated stddev and mean from', Number, 100)
+    this.option('lastpoints', 'Trades for array 2 to be calculated stddev and mean from', Number, 5)
+    this.option('min_periods', 'min_periods', Number, 150)
+    this.option('order_adjust_time', 'Order Adjust Time', Number, 30000)
+    this.option('trend', 'Float number 1-2 would be increasing trend', Number, 1.1)
   },
-
   calculate: function () {
-
   },
-
   onPeriod: function (s, cb) {
-    ema(s, 'trendline', s.options.trendline)
+    ema(s, 'stddev', s.options.stddev)
     var tl1 = []
     var tls = []
-    var tll = []
-    if (s.lookback[s.options.avgpoints + 2000]) {
-      for (let i = 0; i < s.options.avgpoints + 1000; i++) { tl1.push(s.lookback[i].close) }
+    if (s.lookback[s.options.min_periods]) {
+      for (let i = 0; i < s.options.avgpoints + 10; i++) { tl1.push(s.lookback[i].close) }
```

Comment:
if you use and internal iterator, instead of this external, you may could prevent the typerror (see other command). 
if you also add a consistency check you could skip trading, while not having enough data, instead crashing.